src/scenarios/personalization/ArrayAddFavorites/ArrayAddFavorites.tsx

```diff
@@ -0,0 +1,245 @@
+import {
+  useScenarioHost,
+  VariableSearch,
+} from '@axinom/mosaic-fe-samples-host';
+import { useState } from 'react';
+import {
+  Container,
+  Divider,
+  Form,
+  Grid,
+  Header,
+  Label,
+  Segment,
+} from 'semantic-ui-react';
+import { getApolloClient } from '../../../apollo-client';
+import {
+  getMoviesQuery,
+  getTvShowsQuery,
+  setArrayItemMutation,
+} from './graphql-documents';
+
+interface entity {
+  id: string;
+  title: string;
+  type: string;
+}
+
+export const ArrayAddFavorites: React.FC = () => {
+  const { activeProfile, logger } = useScenarioHost();
+  const [userAccessToken, setUserAccessToken] = useState<string>();
+
+  const [entityArray, setEntityArray] = useState<entity[]>([]);
+  const [entityId, setSelectedEntity] = useState<string>('');
+
+  const fetchAllCatalogItems = async (): Promise<void> => {
+    try {
+      const apolloClient = getApolloClient(
+        new URL('graphql', activeProfile.catalogServiceBaseURL).href,
+      );
+
+      const resultMovies = await apolloClient.query({
+        query: getMoviesQuery,
+        context: {
+          headers: {
+            Authorization: `Bearer ${userAccessToken}`,
+          },
+        },
+        fetchPolicy: 'no-cache',
+      });
+
+      const resultTvShows = await apolloClient.query({
+        query: getTvShowsQuery,
+        context: {
+          headers: {
+            Authorization: `Bearer ${userAccessToken}`,
+          },
+        },
+        fetchPolicy: 'no-cache',
+      });
+
+      const moviesArray = resultMovies.data.movies.nodes;
```

Comment:
We can do this as follows. No need to have the for loop then (we use `result` now because we use only a single query now. so `resultMovies` and `resultTvShows` are not there anymore)

```
const moviesArray = result.data.movies.nodes.map(movie -> {...movie, type: 'Movie'};
const tvShowsArray = result.data.tvshows.nodes.map(tvshow -> {...tvshow, type: 'TV Show'};
```